File: m/rls/rlb/stdif.c
/* file -  stdif - differ -- compare limited strings */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_dif - differ -- compare limited strings */
st_dif(
char *mod ,
char *tar ,
size_t lim )
{ for(;;)  {
    if( lim-- == 0)return ( 0 );
    if( *mod != *tar)break;
    if( *tar == 0)return ( 0 );
    ++mod, ++tar;
  } 
  return ( (*mod > *tar) ? 1: -1);
} 

File: m/rls/stdif.r
file	stdif - differ -- compare limited strings
include rid:rider
include rid:stdef

code	st_dif - differ -- compare limited strings

  func	st_dif
	mod : * char
	tar : * char
	lim : size			; compare limit
  is	repeat				;
	   reply 0 if lim-- eq		; all done
	   quit if *mod ne *tar		; different strokes
	   reply 0 if *tar eq		; end of string
	   ++mod, ++tar			; next
	end				;
	reply (*mod gt *tar) ? 1 ?? -1	; difference
  end
